### ISSUE TITLE
Build error[E0282]

### ISSUE THREAD

**Issue by jesselucas** [jmacdonald/amp#81]
When building from source on macOS 10.13.4 I receive the following error:

```
error[E0282]: type annotations needed
  --> /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/app_dirs-1.1.1/src/imp/platform/macos.rs:10:40
   |
10 |         Ok(Component::RootDir.as_ref().into())
   |                                        ^^^^ cannot infer type for `T`

error: aborting due to previous error
```

Toolchain:
```
stable-x86_64-apple-darwin (default)
rustc 1.25.0 (84203cac6 2018-03-25)
```

Found a similar error reported in sccache: https://github.com/mozilla/sccache/issues/233

**Comment by jmacdonald** [comment]
You may need to update to the latest Rust stable, which is `1.25` as of this writing.

**Comment by jesselucas** [comment]
I've attempted with:

`rustc 1.27.0-nightly (7360d6dd6 2018-04-15)` &
` rustc 1.25.0 (84203cac6 2018-03-25)`

with the same results. I could have messed something else up though 🙏

**Comment by mseri** [comment]
I have got the same issue with rust 1.25.0 on osx. But it seems an `app_dirs` issue, not an amp one: 
```
error[E0619]: the type of this value must be known in this context
  --> .brew_home/.cargo/registry/src/github.com-1ecc6299db9ec823/app_dirs-1.1.1/src/imp/platform/macos.rs:10:40
   |
10 |         Ok(Component::RootDir.as_ref().into())
   |                                        ^^^^

error: aborting due to previous error

error: Could not compile `app_dirs`.
```

**Comment by mseri** [comment]
Indeed, it's a problem with `app_dirs 1.1.1` and `rustc >= 1.25.0`. It has been fixed in `app_dirs >= 1.2.0`
